frontend/src/metabase-lib/lib/queries/StructuredQuery.ts

```diff
@@ -1743,6 +1742,10 @@ class NestedStructuredQuery extends StructuredQuery {
     );
   }
 
+  rootQuery(): StructuredQuery {
```

Comment:
Basically the idea is to make `rootQuery` return a query with `source-query` updated if it was changed. Currently it doesn't work and returned an unmodified root query.